internal/utils/grpc.go

```diff
@@ -18,6 +19,22 @@ func CloseClientConnection(connection *grpc.ClientConn) {
 	}
 }
 
+func NewGRPCServer() *grpc.Server {
+	return grpc.NewServer(
+		grpc.KeepaliveEnforcementPolicy(keepalive.EnforcementPolicy{
+			MinTime:             45 * time.Second, // If a client pings more than once every 45 seconds, terminate the connection
+			PermitWithoutStream: true,             // Allow pings even when there are no active streams
+		}),
+		grpc.KeepaliveParams(keepalive.ServerParameters{
+			MaxConnectionIdle:     math.MaxInt64,   // If a client is idle for INFINITE seconds, send a GOAWAY.
+			MaxConnectionAge:      math.MaxInt64,   // If any connection is alive for more than INIFINITE seconds, send a GOAWAY.
+			MaxConnectionAgeGrace: math.MaxInt64,   // Allow INIFNITE seconds for pending RPCs to complete before forcibly closing connections.
+			Time:                  2 * time.Hour,   // Ping the client if it is idle for 2 Hours to ensure the connection is still active.
+			Timeout:               5 * time.Minute, // Wait 5 minutes for the ping ack before assuming the connection is dead.
```

Comment:
I'm not sure about that. If a service isn't responding for 5min I wouldn't expect any answer at all. The default value is 20seconds here. Is there a reason why we want to elevate it to 5min?